### ISSUE TITLE
Could not load type 'MediaBrowser.Model.Serialization.IJsonSerializer' 

### ISSUE THREAD

**Issue by rpardini** [jellyfin/jellyfin-plugin-kodisyncqueue#36]
On Jellyfin 10.7.0 nightly ((20210119.9-unstable))

```
[03:51:56] [INF] [1] Main: Jellyfin version: 10.7.0
....
[03:51:56] [INF] [1] Main: Operating system: Linux
[03:51:56] [INF] [1] Main: Architecture: X64
[03:51:56] [INF] [1] Main: 64-Bit Process: True
[03:51:56] [INF] [1] Main: User Interactive: True
[03:51:56] [INF] [1] Main: Processor count: 8
[03:51:56] [INF] [1] Main: Program data path: /config/data
[03:51:56] [INF] [1] Main: Web resources path: /usr/share/jellyfin/web
[03:51:56] [INF] [1] Main: Application directory: /usr/lib/jellyfin/bin/
[03:51:56] [INF] [1] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: /config/cache
....
[03:51:56] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loading assemblies
[03:51:56] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly LiteDB, Version=4.1.4.0, Culture=neutral, PublicKeyToken=4ee40123013c9f27 from /config/data/plugins/Kodi Sync Queue_6.0.0.0/LiteDB.dll
[03:51:56] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.KodiSyncQueue, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null from /config/data/plugins/Kodi Sync Queue_6.0.0.0/Jellyfin.Plugin.KodiSyncQueue.dll
[03:51:59] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Error creating Jellyfin.Plugin.KodiSyncQueue.Plugin
System.TypeLoadException: Could not load type 'MediaBrowser.Model.Serialization.IJsonSerializer' from assembly 'MediaBrowser.Model, Version=10.7.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
   at System.Reflection.RuntimeConstructorInfo.get_Signature()
   at System.Reflection.RuntimeConstructorInfo.GetParametersNoCopy()
   at System.Reflection.RuntimeConstructorInfo.GetParameters()
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.Plugins.PluginManager.CreatePluginInstance(Type type)
[03:51:59] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Plugin /config/data/plugins/Kodi Sync Queue_6.0.0.0 has been disabled.
[03:51:59] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: OMDb 10.7.0.0
[03:51:59] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: MusicBrainz 10.7.0.0
[03:51:59] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: AudioDB 10.7.0.0
```

Let me know if you need anything else. Thanks!

**Comment by sjiampojamarn** [comment]
same here

```
[2021-01-30 09:29:20.512 -08:00] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Error creating "Jellyfin.Plugin.KodiSyncQueue.Plugin"
System.TypeLoadException: Could not load type 'MediaBrowser.Model.Serialization.IJsonSerializer' from assembly 'MediaBrowser.Model, Version=10.7.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
   at System.Reflection.RuntimeConstructorInfo.get_Signature()
   at System.Reflection.RuntimeConstructorInfo.GetParametersNoCopy()
   at System.Reflection.RuntimeConstructorInfo.GetParameters()
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.Plugins.PluginManager.CreatePluginInstance(Type type)
[2021-01-30 09:29:20.537 -08:00] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Plugin "/opt/jellyfin/data/plugins/Kodi Sync Queue_6.0.0.0" has been disabled.
```

**Comment by morbificagent** [comment]
And one guy more in the list :-(

Same here

**Comment by mcarlton00** [comment]
Try uninstalling the plugin, restarting your server, and then reinstalling it.  There's a bug that's causing plugins to fail on upgrade to one of the server RC versions, so if that doesn't work you may need to manually delete the plugin from the filesystem and then repeat these steps.

**Comment by morbificagent** [comment]
Habe tried it twice some minutes ago.

1. Uninstalled the plugin
2. Files where deleted automatically, manually deleted the plugin config 
3. Installed the plugins again 

-> same error message and plugin disabled

**Comment by rpardini** [comment]
Yes. Reinstalling does not help. Using STABLE Jellyfin it does work though.
I suppose API changes in Jellyfin break the Kodi plugin; if Kodiplugin is fixed for unstable it will break stable?
This is just a preview of the problems when new Jellyfin is released, maybe it will get fixed then.

**Comment by mcarlton00** [comment]
Are you using unstable, or the stable-rc release of the server?

**Comment by morbificagent** [comment]
Its the git version compiled yesterday...

**Comment by mcarlton00** [comment]
Unstable has already moved on and is no longer compatible with the published plugins.  They're designed to work with the changes present in the 10.7 release branch, so you'll need to be running one of the 10.7 release candidates, not latest git

**Comment by Uatschitchun** [comment]
As I'm on unstable and having the same error, will the plugin get updated with the next rc or with final release?

**Comment by mcarlton00** [comment]
No.  Unless a bug is found, the plugin is likely in it's final state for 10.7.0, at least.  Just so it's clear, 10.7 and unstable are not and will never be the same thing again.

* Unstable -> Master branch.  Changes constantly, very prone to breakage.  Plugins usually don't work here, though work is going towards making this nicer
* 10.7 -> Release branch created from master in early December that has bugfixes cherry picked into it

By using unstable, you're effectively using 10.7 + some things that won't be included until 10.8

**Comment by sjiampojamarn** [comment]
Make sense. Thanks for the clarification. Confirmed, the plug-in works with v10.7.0-rc3.
